### DISCUSSION TITLE
How to load the form controls from JSON/NoSQL

### DISCUSSION THREAD

**Discussion by amitnil-dev** [myndpm/open-source#45] [Dynamic Forms]
How do i load controls from JSON file,  say example

  createMatConfig('INPUT', {
          name: 'firstName',
          options: { validators: ['required'] },
          factory: { cssClass: 'col-sm-6 col-md-4' },
          params: { label: 'First Name *' },
        }),

I want to get all these from JSON and load the controls.

**Comment by amitnil-dev** [comment]
this will details better:

I have a strange situation where i need to load all the control from a JSON / NoSql db(mongoDb). I have been playing around with the library. Its great work. I wanted to understand how do I load these parameter from JSON file  to 
  controls: [
        createMatConfig('INPUT', {
          name: 'firstName',
          options: { validators: ['required'] },
          factory: { cssClass: 'col-sm-6 col-md-4' },
          params: { label: 'First Name *' },
        }),
my JSON file will have string parameter like {
  "data": [
 
    {

    "formName": "Form_1",
    "type": "INPUT",
    "name": "'firstName'",
    "options": "{ validators: ['required'] }",
    "factory": "{ cssClass: 'col-sm-6 col-md-4' }",
    "params": "{ label: 'First Name *' }"

    }
]
or is there a better way to do this, a bit new to angular sorry 🙂 🛩️

**Comment by matheo** [comment]
Hi @amitnil-dev!
how do you define your children controls?
or your form is supposed to have only one level of controls?
if so, you need to play Javascript with your data like:
```js
myconfig.controls = json.data.map(
  ({ type, ...config }) => createMatConfig(type, config)
);
```
you're always able to put some functions to process your incoming `config` before passing it to `createMatConfig` :wink:

**Reply by amitnil-dev** [reply]
I i think I got what you mean.  I will play with it. What I was trying to do was create metadata which defines all form controls in a JSON/NoSQL db like mongodb and then load them depending on which form the user requests. I will give it a try and play with it.
Angular burns my head sometimes cause of the alien-extradimension verbose.  😉.

secondly can you suggest me some material/ tutorials where in can learn a bit more indepth. I was a old school javascript back then after i switched to c++/c# etc for several years now. So its kinda confusing for me at the moment. I need some debugging skills for angular perhaps.

cheers mate, appreciate  your response. 
wish me luck.
